### PR TITLE
Organize "Specific references" into clearer tables 

### LINKED ISSUE #3118: Link between Specific references and SCT functions
**Feature justification: Is your feature request related to a problem? Please describe.**

Reference [page](https://spinalcordtoolbox.com/en/stable/overview/references.html#references) of RTD states:
```
If you use SCT in your research or as part of your developments, please always cite the main reference. If you use specific tools such as sct_deepseg_sc or the PAM50 template, please also cite the specific articles listed in specific references.
```

But in the section [Specific References](https://spinalcordtoolbox.com/en/stable/overview/references.html#specific-references): it is maybe not clear enough which paper refers to which SCT function.

**Describe the solution you'd like**

It would be nice to make this link clearer: via a table (SCT function // Paper to cite)? or a simple mention (please cite this article if you used sct_*)

### PR DESCRIPTION
**Pull request by joshuacwnewton**
#### GitHub

- [X] I've given this PR a concise, self-descriptive, and meaningful title
- [X] I've linked relevant issues in the PR body
- [X] I've applied [the relevant labels](https://www.neuro.polymtl.ca/software/contributing#pr_labels) to this PR
- [x] I've assigned a reviewer

#### PR contents

- [ ] I've consulted [SCT's internal developer documentation](https://github.com/neuropoly/spinalcordtoolbox/wiki) to ensure my contribution is in line with any relevant design decisions
- [ ] I've added [relevant tests](https://github.com/neuropoly/spinalcordtoolbox/wiki/Programming%3A-Tests) for my contribution
- [X] I've updated the [relevant documentation](https://github.com/neuropoly/spinalcordtoolbox/wiki/Programming%3A-Documentation) for my changes, including argparse descriptions, docstrings, and ReadTheDocs tutorial pages

## Description

Previously, references for CLI tools and Template/Atlas were provided in an unordered list. It was unclear which references referred to which parts of SCT, so this PR creates tables to make it clearer.

## Linked issues

Fixes #3118.

### CODE REVIEW

**Review by jcohenadad — COMMENTED** [review]
This is brillant! 🏅 
A slight cosmetic (and UX) suggestion: I would remove the bullet point in the table, and only leave one citation (other people are not sure which one to cite). I'll comment on a case-by-case basis

File: documentation/source/overview/references.rst [hunk 1]
```diff
@@ -47,69 +47,66 @@ Main reference
...
-   estimation of partial volume effect. Neuroimage
-   2015 <https://www.ncbi.nlm.nih.gov/pubmed/26099457>`__
--  `De Leener B, Fonov VS, Louis Collins D, Callot V, Stikov N,
-   Cohen-Adad J. PAM50: Unbiased multimodal template of the brainstem
-   and spinal cord aligned with the ICBM152 space. Neuroimage
-   2017. <http://www.sciencedirect.com/science/article/pii/S1053811917308686>`__
+The table below provides individual references for novel methods used in SCT's :ref:`command-line-tools`.
+
+.. note::
+   If you are using segmentation tools (``sct_propseg``/``sct_deepseg``) and registration tools (``sct_register_to_template``/``sct_register_multimodal``) together as part of a pipeline, please also consider this reference:
```
Comment (jcohenadad) [inline documentation/source/overview/references.rst:56]: the article is specific to the GM segmentation
```suggestion
   If you are using segmentation tools (``sct_deepseg_gm``/``sct_deepseg``) and registration tools (``sct_register_to_template``/``sct_register_multimodal``) together as part of a pipeline, please also consider this reference:
```

File: documentation/source/overview/references.rst [hunk 2]
```diff
@@ -47,69 +47,66 @@ Main reference
...
+   * - ``sct_deepseg_sc``
+     - * `Gros et al. Automatic segmentation of the spinal cord and intramedullary multiple sclerosis lesions with convolutional neural networks. Neuroimage 2019 <https://www.sciencedirect.com/science/article/pii/S1053811918319578>`__
+   * - ``sct_get_centerline``
+     - * `Gros et al. Automatic spinal cord localization, robust to MRI contrasts using global curve optimization. Med Image Anal 2018 <https://www.sciencedirect.com/science/article/pii/S136184151730186X>`__
+   * - ``sct_label_vertebrae``
+     - * `Ullmann et al. Automatic labeling of vertebral levels using a robust template-based approach. Int J Biomed Imaging 2014 <http://downloads.hindawi.com/journals/ijbi/2014/719520.pdf>`__
+   * - ``sct_propseg``
+     - * `De Leener et al. Robust, accurate and fast automatic segmentation of the spinal cord. Neuroimage 2014 <https://www.ncbi.nlm.nih.gov/pubmed/24780696>`__
+       * `De Leener et al. Automatic segmentation of the spinal cord and spinal canal coupled with vertebral labeling. IEEE Transactions on Medical Imaging 2015 <https://www.ncbi.nlm.nih.gov/pubmed/26011879>`__
```
Comment (jcohenadad) [inline documentation/source/overview/references.rst:78]: I would create a new row for this citation, with the function call: `sct_propseg -CSF`

File: documentation/source/overview/references.rst [hunk 3]
```diff
@@ -47,69 +47,66 @@ Main reference
...
+   * - ``sct_register_multimodal``
+     - * `Cohen-Adad et al. Slice-by-slice regularized registration for spinal cord MRI: SliceReg. Proc ISMRM 2015 <https://www.dropbox.com/s/v3bb3etbq4gb1l1/cohenadad_ismrm15_slicereg.pdf?dl=0>`__
```
Comment (jcohenadad) [inline documentation/source/overview/references.rst:80]: I would create a new row for this citation, and associate it with: `sct_register_multimodal -param algo=slicereg`

File: documentation/source/overview/references.rst [hunk 4]
```diff
@@ -47,69 +47,66 @@ Main reference
...
+       * `De Leener B, Fonov VS, Louis Collins D, Callot V, Stikov N, Cohen-Adad J. PAM50: Unbiased multimodal template of the brainstem and spinal cord aligned with the ICBM152 space. Neuroimage 2017. <http://www.sciencedirect.com/science/article/pii/S1053811917308686>`__
+   * - ``sct_register_to_template``
+     - * `Cohen-Adad et al. Slice-by-slice regularized registration for spinal cord MRI: SliceReg. Proc ISMRM 2015 <https://www.dropbox.com/s/v3bb3etbq4gb1l1/cohenadad_ismrm15_slicereg.pdf?dl=0>`__
```
Comment (jcohenadad) [inline documentation/source/overview/references.rst:83]: I would create a new row for this citation, and associate it with: `sct_register_to_template -param algo=slicereg`

File: documentation/source/overview/references.rst [hunk 5]
```diff
@@ -47,69 +47,66 @@ Main reference
...
+   * - MNI-Poly-AMU template
+     - * `Fonov et al. Framework for integrated MRI average of the spinal cord white and gray matter: The MNI-Poly-AMU template. Neuroimage 2014. <https://www.ncbi.nlm.nih.gov/pubmed/25204864>`__
+   * - White matter atlas
+     - * `Lévy et al. White matter atlas of the human spinal cord with estimation of partial volume effect. Neuroimage 2015 <https://www.ncbi.nlm.nih.gov/pubmed/26099457>`__
+   * - Probabilistic atlas (AMU40)
+     - * `Taso et al. Construction of an in vivo human spinal cord atlas based on high-resolution MR images at cervical and thoracic levels: preliminary results. MAGMA, Magn Reson Mater Phy 2014 <https://www.ncbi.nlm.nih.gov/pubmed/24052240>`__
+       * `Taso et al. A reliable spatially normalized template of the human spinal cord–Applications to automated white matter/gray matter segmentation and tensor-based morphometry (TBM) mapping of gray matter alterations occurring with age. Neuroimage 2015 <https://www.ncbi.nlm.nih.gov/pubmed/26003856>`__
+   * - Motivations for template-based SC analysis, and proposal of method for predicting spinal root locations
```
Comment (jcohenadad) [inline documentation/source/overview/references.rst:108]:
```suggestion
   * - Spinal levels
```

File: documentation/source/overview/references.rst [hunk 6]
```diff
@@ -47,69 +47,66 @@ Main reference
...
+   * - Template/atlas
+     - References
+   * - PAM50 template
+     - * `De Leener B, Fonov VS, Louis Collins D, Callot V, Stikov N, Cohen-Adad J. PAM50: Unbiased multimodal template of the brainstem and spinal cord aligned with the ICBM152 space. Neuroimage 2017. <http://www.sciencedirect.com/science/article/pii/S1053811917308686>`__
+   * - MNI-Poly-AMU template
+     - * `Fonov et al. Framework for integrated MRI average of the spinal cord white and gray matter: The MNI-Poly-AMU template. Neuroimage 2014. <https://www.ncbi.nlm.nih.gov/pubmed/25204864>`__
+   * - White matter atlas
+     - * `Lévy et al. White matter atlas of the human spinal cord with estimation of partial volume effect. Neuroimage 2015 <https://www.ncbi.nlm.nih.gov/pubmed/26099457>`__
+   * - Probabilistic atlas (AMU40)
+     - * `Taso et al. Construction of an in vivo human spinal cord atlas based on high-resolution MR images at cervical and thoracic levels: preliminary results. MAGMA, Magn Reson Mater Phy 2014 <https://www.ncbi.nlm.nih.gov/pubmed/24052240>`__
```
Comment (jcohenadad) [inline documentation/source/overview/references.rst:106]: it overlaps with the 2nd citation so i would just remove it
```suggestion
```

**Comment by joshuacwnewton** [comment]
Glad to see that `sphinx.fail_on_warning` caught my formatting typo. Working as intended! :wink:

**Review by jcohenadad — APPROVED** [review]
Awesome contribution, thank you!

**Pull request merged**